enacts/wat_bal/tests/test_agronomy.py

```diff
@@ -138,4 +138,31 @@ def test_api_runoff():
     )
 
     assert np.allclose (api, [np.nan, 0.5, 2, 3.5, 5], equal_nan=True)
-    assert np.allclose(runoff, [0, 1 + 1*2 + 1*2**2, 1 + 2*3 + 3*3**2, -2 + 0*4 + 1*4**2])
\ No newline at end of file
+    assert np.allclose(runoff, [0, 1 + 1*2 + 1*2**2, 1 + 2*3 + 3*3**2, -2 + 0*4 + 1*4**2])
+
+
+def test_solar_radiation():
+    t = xr.DataArray(
+        pd.date_range(start="2000-06-21", end="2000-12-21", freq="7D"),
+        dims=["T"]
+    )
+    doy = t.dt.dayofyear
+    lat = xr.DataArray(np.arange(40, 80, 10), dims=["Y"])
+    lat = lat * np.pi / 180
+    ra = agronomy.solar_radiation(doy, lat).dropna("Y")
+
+    # In higher latitudes:
+    # RA decreases with lenght of days
+    assert (ra.diff("T") <= 0).all()
+    # Ra decreases with latitude
+    assert (ra.diff("Y") <= 0).all()
+
+
+def test_hargreaves_et_ref():
+    et_ref = agronomy.hargreaves_et_ref(
+        xr.DataArray(32),
+        xr.DataArray(5),
+        xr.DataArray(40)
+    )
+
+    assert et_ref == 0.0023 * (32 + 17.8) * np.sqrt(5) * 0.408 * 40
```

Comment:
The red line in a circle indicates that the file doesn't have a newline at the end. This results in spurious diffs when the file is later extended; see line 141 in this diff for an example. You can make future diffs more readable by adding a newline here now.